finesse/gui/measure_script/script_edit_dialog.py

```diff
@@ -0,0 +1,414 @@
+    def delete_all(self) -> None:
+        """Delete all instructions from the table."""
+        self.model.beginRemoveRows(QModelIndex(), 0, len(self.sequence) - 1)
+        self.sequence.clear()
+        self.model.endRemoveRows()
+
+
+class SequenceButtons(QWidget):
+    """Buttons for changing the sequence of instructions."""
+
+    def __init__(self, sequence: SequenceWidget) -> None:
+        """Create a new SequenceButtons."""
+        super().__init__()
+
+        add_buttons = AddButtons(sequence)
+        change_buttons = ChangeButtons(sequence)
+
+        layout = QVBoxLayout()
+        layout.addWidget(add_buttons)
+        layout.addWidget(change_buttons)
+        self.setLayout(layout)
+
+
+class AddButtons(QGroupBox):
+    """Buttons for adding new instructions."""
+
+    def __init__(self, sequence: SequenceWidget) -> None:
+        """Create a new AddButtons."""
+        super().__init__("Add instruction")
+        self.sequence = sequence
+
+        layout = QVBoxLayout()
+
+        # The number of times to repeat this measurement
+        self.count = CountWidget("Measurements")
+        layout.addWidget(self.count)
+
+        # Add buttons for preset angles (e.g. zenith, nadir, etc.)
+        self.group = QButtonGroup()
+        self.group.buttonClicked.connect(self._preset_clicked)  # type: ignore
+        for preset in ANGLE_PRESETS:
+            btn = QPushButton(preset.upper())
+            self.group.addButton(btn)
+            layout.addWidget(btn)
+
+        # Add button and spinbox for going to a specific angle
+        self.angle = QSpinBox()
+        self.angle.setMinimum(0)
+        self.angle.setMaximum(359)
+        goto = QPushButton("GOTO")
+        goto.clicked.connect(self._goto_clicked)  # type: ignore
+        layout.addWidget(self.angle)
+        layout.addWidget(goto)
+
+        self.setLayout(layout)
+
+    def _goto_clicked(self) -> None:
+        self.sequence.add_instruction(float(self.angle.value()), self.count.value())
+
+    def _preset_clicked(self, btn: QPushButton) -> None:
+        self.sequence.add_instruction(btn.text().lower(), self.count.value())
+
+
+class ChangeButtons(QGroupBox):
+    """Buttons for modifying existing measure instructions."""
+
+    def __init__(self, sequence: SequenceWidget) -> None:
+        """Create a new ChangeButtons."""
+        super().__init__("Modify instructions")
+
+        up = QPushButton("Up")
+        up.clicked.connect(sequence.move_selected_up)  # type: ignore
+
+        down = QPushButton("Down")
+        down.clicked.connect(sequence.move_selected_down)  # type: ignore
+
+        delete = QPushButton("Delete")
+        delete.clicked.connect(sequence.delete_selected)  # type: ignore
+
+        clear = QPushButton("Clear")
+        clear.clicked.connect(sequence.delete_all)  # type: ignore
+
+        layout = QVBoxLayout()
+        layout.addWidget(up)
+        layout.addWidget(down)
+        layout.addWidget(delete)
+        layout.addWidget(clear)
```

Comment:
This button is too close to the `delete` one. When trying the thing, after writing a few lines, I click that one by mistake and all was gone wit question. I would, add a confirmation dialog because otherwise the loss of work can be drastic.

Reply:
Good idea. I've added a confirmation dialog.

I wasn't sure whether the button should live somewhere else, but then I couldn't think of where else to put it. The dialog makes it less dangerous anyways though.